apis.py

```diff
@@ -29,17 +29,20 @@ def get_coins():
 def get_pool_size(pool, coin):
     if pool not in CACHED_POOL_SIZES or time.time() > CACHED_POOL_SIZES[pool][0] + API_WAIT_SECONDS:
         try:
-            api_key = CONFIG['pools'][pool]['api_key']
-            r = requests.get(
-                CONFIG['pools'][pool]['coins'][coin]['pool_api']['url'].replace('$API_KEY', api_key)
-            ).json()
+            request = CONFIG['pools'][pool]['coins'][coin]['pool_api']['url']
+            if contains_api_key(request):
+                    api_key = USER_CONFIG['pools'][pool]['api_key']
+                    request.replace('$API_KEY', api_key)
+            r = requests.get(request).json()
             for key in CONFIG['pools'][pool]['coins'][coin]['pool_api']['hashrate']:
                 r = r[key]
             CACHED_POOL_SIZES[pool] = (time.time(), r)
-        except Exception:
-            print('Failed to get pool size of', pool, 'for', coin)
+        except requests.exceptions.RequestException:
```

Comment:
Nice, I think there's the possibility also that the request returns but it's a message instead of a JSON object and it fails at parsing. It looks like it's json.decoder.JSONDecodeError (I tested this by getting google.com and calling .json() on the response). I think that's what happens if the request gets rejected but the url isn't down, so it might be worth also catching here